src/test/java/org/mule/service/http/impl/service/server/grizzly/HttpRequestMultipleAuthenticationMethodsSupportedTestCase.java

```diff
@@ -0,0 +1,151 @@
+
+    if (StringUtils.isEmpty(authHeader)) {
+      responseBuilder.statusCode(UNAUTHORIZED.getStatusCode())
+          .addHeader(WWW_AUTHENTICATE, BASIC)
+          .addHeader(WWW_AUTHENTICATE, NTLM.name())
+          .addHeader(WWW_AUTHENTICATE,
+                     DIGEST + " qop=\"auth\",algorithm=MD5-sess," +
+                         "nonce=\"" + NONCE + "\",charset=utf-8,realm=\"INT\"");
+    } else if (authType.length() > 0) {
+      chosenAuthMethod = authType;
+      responseBuilder.statusCode(OK.getStatusCode())
+          .reasonPhrase(OK.getReasonPhrase());
+    } else {
+      responseBuilder.statusCode(UNAUTHORIZED.getStatusCode());
+    }
+    return responseBuilder.build();
+  }
+
+  public HttpRequestOptions getOptions(HttpAuthenticationBuilder authenticationBuilder) {
+    HttpAuthentication authentication = authenticationBuilder.build();
+    return HttpRequestOptions.builder().authentication(authentication).build();
+  }
+
+  public HttpRequest getRequest() {
+    return HttpRequest.builder().method(GET).uri(getUri()).build();
+  }
+
+  public String getRequestAuthType(String authHeader) {
+    return StringUtils.isEmpty(authHeader) ? "" : authHeader.substring(0, authHeader.indexOf(" "));
+  }
+
+  @Before
+  public void setUpClient() {
+    client = service.getClientFactory().create(clientBuilder.build());
+    client.start();
+  }
+
+  @After
+  public void stopClient() {
+    if (client != null) {
+      client.stop();
+    }
+  }
+
+  @Test
+  public void onMultipleAuthenticationMethodsSupportedLocallySetMethodIsChosenDigest() {
+    sendRequestAssertCorrectAuthMethodChosen(HttpAuthentication.digest(TEST_USER, TEST_PASS), DIGEST);
+  }
+
+  @Test
+  public void onMultipleAuthenticationMethodsSupportedLocallySetMethodIsChosenNtlm() {
+    sendRequestAssertCorrectAuthMethodChosen(HttpAuthentication.ntlm(TEST_USER, TEST_PASS), NTLM.name());
+  }
+
+  @Test
+  public void onMultipleAuthenticationMethodsSupportedLocallySetMethodIsChosenBasic() {
+    sendRequestAssertCorrectAuthMethodChosen(HttpAuthentication.basic(TEST_USER, TEST_PASS), BASIC);
+  }
+
+  protected void sendRequestAssertCorrectAuthMethodChosen(HttpAuthenticationBuilder authBuilder, String authMethod) {
+    HttpRequestOptions options = getOptions(authBuilder);
+    HttpRequest request = getRequest();
+    client.sendAsync(request, options).whenComplete(
+                                                    (response, exception) -> responseReference.set(response));
+    pollingProber.check(new ResponseReceivedProbe(responseReference));
+    pollingProber.check(new ResponseSuccessProbe());
+    assertThat(chosenAuthMethod, equalTo(authMethod));
+  }
+
+  private class ResponseSuccessProbe implements Probe {
+
+    @Override
+    public boolean isSatisfied() {
+      return responseReference.get().getStatusCode() == OK.getStatusCode();
+    }
+
+    @Override
+    public String describeFailure() {
+      return "Authorization method chosen should be Basic!";
```

Comment:
You'll be saying it needs to be `Basic` on all tests. You need to parameterize this and use the `authMethod` data to instantiate the `Probe`.